app.js

```diff
@@ -153,12 +180,18 @@ function on_video_played() {
 
 function loop_to_check_player_visibility() {
     console.log("[Better Mobile Youtube] checking if player is visible");
-    videoElement = document.querySelector('video');
-
-    if (videoElement.checkVisibility()) {
-        on_video_played();
+    videoElement = document.querySelector('.html5-main-video');
+
+    if (videoElement) {
+        if (videoElement.checkVisibility()) {
+            console.log("[Better Mobile Youtube] Player is visible");
+            on_video_played(videoElement);           
+        } else {
+            setTimeout(loop_to_check_player_visibility, 150);
+            console.log("[Better Mobile Youtube] Player is not visible");
+        }
     } else {
-        setTimeout(loop_to_check_player_visibility, 1000);
+        setTimeout(loop_to_check_player_visibility, 150);
```

Comment:
just a concern, looping this fast will it cause any issues on low end devices.
the reason i looped ever sec was so under the impression that looping fast may cause high cpu usage on low end devices.

Reply:
I will see about this.

Please wait

Reply:
After some testing with throttling, I've determined that, at least in this case and those like it, the performance hit is negligible.

![Screenshot from 2024-06-29 19-00-39](https://github.com/tusx/better-mobile-youtube/assets/82143316/abfb3b5d-78b6-4d64-a71c-dcf27d42ec2b)

This is with my laptops CPU throttled to 20 times slower than usual. This function only takes up 0.3-0.5% of the processing time.

While having 50 functions like this one will create a hit that might be noticeable, as long as we keep things reasonable, it should be fine.